### PR TITLE
pydax_initialization should localize the download URLs

### PR DESCRIPTION
**Pull request by xuhdev**
This change localizes the URLs in the initialization so that the
reliance on remote hosts is minimized.

After this change, all tests can run without Internet connections after
the first run (which downloads relevant datasets locally). This can
potentially speed up tests if the Internet connections are not very
fast.

---

## Checklist:

<details open>

<summary></summary>

- Does this pull request close an issue? We encourage you to open an issue first if this pull request (PR) is not a
  minor change.
  - [ ] No
    - [ ] The change in this pull request is minor.
  - [x] Yes
    - Close #{internal}

For the following questions, only check the boxes that are applicable.

- [ ] Ensure that the pull request text above the horizontal line is descriptive.
- [ ] Add/update relevant tests.
- [ ] Add/update relevant documents.

- Do you have triage permission of this repository?
  - [ ] No
    - You are good.
  - [x] Yes
    - Does this pull request close an issue?
      - [ ] No, this pull request also acts as an issue by itself.
        - [ ] Assign yourself.
        - [ ] Label this pull request.
        - [ ] Put a milestone.
        - [ ] Put this pull request under "In Progress" or "Under Review" pipeline on ZenHub.
        - [ ] Put this pull request under the appropriate epic on ZenHub.
        - [ ] Put an estimate on ZenHub.
        - [ ] Add dependency relationship with other issues/pull requests on ZenHub.
      - [x] Yes, this pull request addresses an issue and does not act as an issue.
        - [x] Connect this pull request to the underlying issue on ZenHub.
        - [x] DON'T:
          - DON'T assign yourself or anyone else.
          - DON'T label.
          - DON'T put a milestone.
          - DON'T put this pull request under any epic on ZenHub.
          - DON'T put an estimate on ZenHub.
    - [x] When the pull request is ready, request review.
</details>

### CODE REVIEW

**Comment by xuhdev** [comment]
In terms of CI: I just reran our master branch CI. While most running time don't seem to change much, that of the bottleneck one -- MacOS test -- has been reduced from 3m33s to 1m47s.

**Comment by xuhdev** [comment]
After #94, `pydax_initialization` should also be used for doctest in `_high_level.py`.

File: tests/conftest.py [hunk 1]
```diff
@@ -195,25 +202,28 @@ def _download_dataset_impl(name, version):
 
 
 @pytest.fixture(scope='session')
-def _schema(_loaded_schemata, _download_dataset, dataset_base_url) -> Callable[[str, str], SchemaDict]:
-    "Utility function for generating schema fixtures with its downloading URL modified to the local HTTP URL."
+def schema_localized_url(_loaded_schemata, _download_dataset, dataset_base_url) -> Callable[[str, str], SchemaDict]:
+    "Utility function fixture for generating schema fixtures with its downloading URL modified to the local HTTP URL."
```
Comment (edwardleardi) [inline tests/conftest.py]:
```suggestion
    "Utility function fixture for generating schema fixtures with its downloading URL modified to the local HTTPS URL."
```

File: tests/test_high_level.py [hunk 2]
```diff
@@ -196,12 +196,14 @@ def test_loading_undownloaded(self, tmp_path):
...
-    def test_human_param(self, loaded_schemata):
+    def test_human_param(self):
+        "Test the ``human`` parameter of ``get_dataset_metadata``."
+
         name, version = 'gmb', '1.0.2'
 
         gmb_metadata = get_dataset_metadata(name, version=version)
-        dataset_schema = loaded_schemata.schemata['datasets'].export_schema('datasets', name, version)
-        license_schema = loaded_schemata.schemata['licenses'].export_schema('licenses')
+        dataset_schema = _get_schemata().schemata['datasets'].export_schema('datasets', name, version)
```
Comment (edwardleardi) [inline tests/test_high_level.py]: Why are we changing from `loaded_schemata` to `_get_schemata` here?

**Review by edwardleardi — COMMENTED** [review]
Can we also add more detail to the `_download_dataset` fixture docstring? e.g. info like that these dataset downloads are cached for the session after being downloaded the first time, and that this is to download the dataset locally so later tests that need to download a dataset can download from our local cache instead of remotely again.

File: tests/test_high_level.py (hunk 2 above)
Comment (xuhdev) [reply tests/test_high_level.py]: `loaded_schemata` is different from `_get_schemata`, they just happened to be the same before. Here, we are verifying the description should match that of `export_schemata()`. See my additional comments under `loaded_schemata`.

Comment (xuhdev) [reply tests/test_high_level.py]: I changed these to `export_schemata()`

Comment (edwardleardi) [reply tests/test_high_level.py]: ah ok I see now, thanks!

**Review by edwardleardi — APPROVED** [review]

**Pull request merged**